presentation/src/main/java/me/androidbox/presentation/HomeActivity.kt

```diff
@@ -21,13 +26,39 @@ class HomeActivity : ComponentActivity() {
         super.onCreate(savedInstanceState)
         val homeViewModel by viewModels<HomeViewModel>()
 
+        val uploadEventWorkerRequest = OneTimeWorkRequestBuilder<UploadEventWorker>()
+            .setConstraints(
+                Constraints.Builder()
+                    .setRequiredNetworkType(
+                        NetworkType.CONNECTED).build()
+            ).build()
+
+        val workManager = WorkManager.getInstance(applicationContext)
+
         installSplashScreen().apply {
             this.setKeepOnScreenCondition {
                 homeViewModel.authenticationState.value == null
             }
         }
 
         setContent {
+            val workInfos by workManager.getWorkInfosForUniqueWorkLiveData("upload_event").observeAsState()
+            
+            val downloadInfo = remember(key1 = workInfos) {
+                workInfos?.find { workInfo ->
+                    workInfo.id == uploadEventWorkerRequest.id
+                }
+            }
+
+            val eventState by derivedStateOf {
+                val downloadEvent = downloadInfo?.outputData?.getString("")?.toUri()
+            }
```

Comment:
This `derivedStateOf` does nothing, since you don't return any state in the block (and if you do, it would need to be wrapped with `mutableStateOf()`)

Reply:
@philipplackner I was just trying this out and forgot to remove it.